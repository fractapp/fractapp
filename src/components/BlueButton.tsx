import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

/**
 * @category Components
 */
export const BlueButton = ({
  text,
  height,
  disabled = false,
  width = '100%',
  onPress,
}: {
  text: string;
  height: number;
  disabled?: boolean;
  width?: string;
  onPress: () => void;
}) => {
  return (
    <TouchableOpacity
      style={[
        styles.button,
        {
          height: height,
          width: width,
          backgroundColor: disabled ? '#99E4FF' : '#2AB2E2',
        },
      ]}
      disabled={disabled}
      onPress={onPress}>
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  buttonText: {
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'white',
  },
  button: {
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
