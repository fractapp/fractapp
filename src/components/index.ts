export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
export {TabMenu} from './TabMenu';
export {ChatShortInfo} from './ChatShortInfo';
export {PaymentMsg} from './PaymentMsg';
export {Contact} from './Contact';
export {Receiver} from './Receiver';
export {AmountInput} from './AmountInput';
export {AmountValue} from './AmountValue';
export {ReceiverWithEnterAddress} from './ReceiverWithEnterAddress';
