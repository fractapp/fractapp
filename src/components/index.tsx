export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
