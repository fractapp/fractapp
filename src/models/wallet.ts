import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
