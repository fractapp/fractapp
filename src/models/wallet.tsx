import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
