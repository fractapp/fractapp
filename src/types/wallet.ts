import {ImageSourcePropType} from 'react-native';

/**
 * @category Models
 */
export enum Currency {
  Polkadot = 0,
  Kusama,
}

/**
 * get logo for wallet
 * @category Models
 */
export function getLogo(currency: Currency) {
  let logo: ImageSourcePropType;
  switch (currency) {
    case Currency.Polkadot:
      logo = require('assets/img/dot.png');
      break;
    case Currency.Kusama:
      logo = require('assets/img/kusama.png');
      break;
    default:
      throw new Error('invalid currency');
  }
  return logo;
}

/**
 * check logo without border
 * @category Models
 */
export function withoutBorder(currency: Currency): boolean {
  return currency === Currency.Polkadot || currency === Currency.Kusama;
}

/**
 * get symbol for wallet
 * @category Models
 */
export function getSymbol(currency: Currency) {
  let symbol = '';
  switch (currency) {
    case Currency.Polkadot:
      symbol = 'DOT';
      break;
    case Currency.Kusama:
      symbol = 'KSM';
      break;
    default:
      throw new Error('invalid currency');
  }
  return symbol;
}

/**
 * @category Models
 */
export class Wallet {
  name: string;
  address: string;
  balance: number;
  planks: string;
  currency: Currency;
  usdValue: number;

  constructor(
    name: string,
    address: string,
    currency: Currency,
    balance: number,
    planks: string,
    price: number,
  ) {
    this.name = name;
    this.address = address;
    this.balance = balance;
    this.planks = planks;
    this.currency = currency;

    this.usdValue = +(balance * price).toFixed(2);
  }
}
