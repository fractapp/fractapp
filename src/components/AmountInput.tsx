import React, { useEffect, useRef, useState } from 'react';
import { ActivityIndicator, Image, StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
import { getSymbol } from 'types/wallet';
import { Adaptors, TransferValidationArgs } from 'adaptors/adaptor';
import BN from 'bn.js';
import MathUtils from 'utils/math';
import math from 'utils/math';
import StringUtils from 'utils/string';
import { Account } from 'types/account';
import { EnterAmountInfo } from 'types/inputs';
import { TxAction } from 'types/transaction';
import { EnterAmountArgs } from 'types/message';

/**
 * Text input for amount
 * @category Components
 */
export const AmountInput = ({
  account,
  args,
  isChatBotRequest,
  price,
  onChangeValues,
  onSetLoading,
  defaultValue,
  defaultUsdMode,
  isChatBotLimit,
  width = '100%',
}: {
  account: Account;
  args: EnterAmountArgs | TransferValidationArgs;
  isChatBotRequest: boolean;
  price: number;
  onChangeValues: (
    newEnterAmount: EnterAmountInfo
  ) => void;
  onSetLoading: (isLoading: boolean) => void;
  defaultValue: string;
  defaultUsdMode: boolean;
  isChatBotLimit: boolean;
  width?: string;
}) => {
  const api = Adaptors.get(account.network)!;
  const textInputRef = useRef<TextInput>(null);

  const [calculatedFee, setCalculatedFee] = useState(isChatBotRequest ? new BN((args as EnterAmountArgs).fee) : null);
  const limit = isChatBotLimit ? new BN((args as EnterAmountArgs).limit!) : new BN(account.balance.transferable);

  const [errorText, setErrorText] = useState<string>('');

  const [isFeeLoading, setFeeLoading] = useState<boolean>(false);
  const [value, setValue] = useState<string>(defaultValue);
  const [forceValue, setForceValue] = useState<string | null>(null);
  const [usdMode, setUsdMode] = useState<boolean>(defaultUsdMode);

  const [isLoadingEnd, setLoadingEnd] = useState<boolean>(false);
  const [isRecall, setRecall] = useState<boolean>(false);

  const [enterAmountInfo, setEnterAmountInfo] = useState<EnterAmountInfo>({
    value: defaultValue,
    alternativeValue: 0,
    usdValue: 0,
    usdFee: 0,
    planksValue: '0',
    planksFee: '0',
    isUSDMode: defaultUsdMode,
    isValid: true,
  });

  const calculateFee = async (sender: string, value: BN): Promise<BN | undefined> => {
    if (calculatedFee == null) {
      const fee = await api.calculateTransferFee(account.address, value, (args as TransferValidationArgs).receiver);
      if (fee === undefined) {
        return undefined;
      }
      setCalculatedFee(fee);
      return fee;
    } else {
      return calculatedFee;
    }
  };

  const calculateValues = (value: string, isForce: boolean) => {
    if (isFeeLoading && !isForce) {
      setRecall(true);
      return;
    }

    console.log('calculate value');

    const newEnterAmount: EnterAmountInfo = resetValues(true, '');
    onChangeValues(newEnterAmount);
    newEnterAmount.value = value;
    newEnterAmount.isUSDMode = usdMode;

    setFeeLoading(true);

    if (value.length > 0) {
      const v = parseFloat(value);
      const p = new BN(math.convertToPlanck(value, api.decimals));
      if (
        (newEnterAmount.isUSDMode && (isNaN(v) || v <= 0)) ||
        (!newEnterAmount.isUSDMode && p.cmp(new BN(0)) <= 0)
      ) {
        const emptyValue = resetValues(false, '');
        onChangeValues(emptyValue);
        setEnterAmountInfo(emptyValue);
        setLoadingEnd(true);
        return;
      }
    }

    newEnterAmount.isValid = true;
    setErrorText('');

    if (newEnterAmount.isUSDMode) {
      const v = parseFloat(value);
      const usd = isNaN(v) ? 0 : v;
      newEnterAmount.usdValue = usd;

      const p = MathUtils.calculatePlanksValue(
        usd,
        api.decimals,
        price,
      );
      newEnterAmount.planksValue = p.toString();
    } else {
      const p = new BN(math.convertToPlanck(value, api.decimals));
      newEnterAmount.planksValue = p.toString();

      newEnterAmount.usdValue = MathUtils.calculateUsdValue(
        p,
        api.decimals,
        price,
      );
    }

    const planksValue = new BN(newEnterAmount.planksValue);

    calculateFee(account.address, planksValue).then(async (fee) => {
      if (fee === undefined) {
        const emptyValue = resetValues(false, 'Invalid connection. Please, try again.'); //TODO go to string
        onChangeValues(emptyValue);
        setEnterAmountInfo(emptyValue);
        setErrorText('Invalid connection. Please, try again.');
        setLoadingEnd(true);
        return;
      }
      newEnterAmount.planksFee = fee.toString();
      const validateInfo = await validateParams(newEnterAmount.value, planksValue, newEnterAmount.usdValue, newEnterAmount.isUSDMode, fee);
      if (!validateInfo.isValid) {
        const emptyValue = resetValues(validateInfo.isBeforeResetValid, validateInfo.err);
        onChangeValues(emptyValue);
        setEnterAmountInfo(emptyValue);
        setErrorText(validateInfo.err);
        setLoadingEnd(true);
        return;
      }

      newEnterAmount.alternativeValue = newEnterAmount.isUSDMode
        ? MathUtils.convertFromPlanckToViewDecimals(
          planksValue,
          api.decimals,
          api.viewDecimals,
          true,
        ) : newEnterAmount.usdValue;

      newEnterAmount.usdFee = await MathUtils.calculateUsdValue(
        fee,
        api.decimals,
        price,
      );

      setErrorText('');

      console.log('set');
      setEnterAmountInfo(newEnterAmount);
      onChangeValues(newEnterAmount);
      setLoadingEnd(true);
    });
  };

  useEffect(() => {
    if (value === '') {
      const emptyValue = resetValues(true, '');
      onChangeValues(emptyValue);
      setEnterAmountInfo(emptyValue);
      setLoadingEnd(true);
      return;
    }
    calculateValues(value, false);
  }, [value]);
  useEffect(() => {
    console.log('forceValue: ' + forceValue);
    if (forceValue == null) {
      return;
    }
    calculateValues(forceValue, true);
  }, [forceValue]);

  useEffect(() => {
    onSetLoading(isFeeLoading);
  }, [isFeeLoading]);

  useEffect(() => {
    if (!isLoadingEnd) {
      return;
    }

    setLoadingEnd(false);
    if (isRecall) {
      setForceValue(value);
      setRecall(false);
      console.log('recall');
    } else {
      setFeeLoading(false);
      setForceValue(null);
      console.log('fee loading off');
    }
  }, [isLoadingEnd]);

  const resetValues = (isValid: boolean, errorText: string): EnterAmountInfo => {
    const newEnterAmount: EnterAmountInfo = Object.create(enterAmountInfo);

    newEnterAmount.isValid = isValid;
    setErrorText(errorText);

    newEnterAmount.value = '';
    newEnterAmount.planksValue = (new BN(0)).toString();
    newEnterAmount.usdValue = 0;
    newEnterAmount.alternativeValue = 0;
    newEnterAmount.usdFee = 0;
    newEnterAmount.planksFee = '0';

    return newEnterAmount;
  };

  const validateParams = async (value: string, planksValue: BN, usdValue: number, isUSDMode: boolean, fee: BN): Promise<{
    isBeforeResetValid: boolean,
    isValid: boolean,
    err: string
  }> => {
    const plankViewValue = MathUtils.convertFromPlanckToViewDecimals(
      planksValue,
      api.decimals,
      api.viewDecimals,
      true,
    );

    if (
      planksValue!.cmp(new BN(0)) <= 0 ||
      usdValue <= 0 ||
      (isUSDMode && plankViewValue <= 0)
    ) {
      return {
        isBeforeResetValid: !(value.length > 0),
        isValid: false,
        err: '',
      };
    }


    const transferValidation = await api.isTxValid(
      account.address,
      isChatBotRequest ? null : TxAction.Transfer,
      isChatBotRequest ? {
        limit: limit,
      } : (args as TransferValidationArgs),
      planksValue,
      fee
    );

    if (!transferValidation.isOk) {
      return {
        isBeforeResetValid: false,
        isValid: false,
        err: transferValidation.errorMsg,
      };
    }

    return {
      isBeforeResetValid: true,
      isValid: true,
      err: '',
    };
  };


  return (
    <View style={{width: width, alignItems: 'center'}}>
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        <Text style={styles.value}>{usdMode && '$'}</Text>
        <TextInput
          onLayout={() => textInputRef?.current?.focus()}
          style={[
            styles.value,
            {
              paddingLeft: 0,
              paddingRight: 0,
              paddingBottom: 0,
              paddingTop: 0,
              marginTop: 8,
              marginBottom: 8,
            },
          ]}
          ref={textInputRef}
          onChangeText={(text) => {
            setValue(text);
          }}
          textAlign={'center'}
          keyboardType={'decimal-pad'}
          placeholderTextColor={'#BFBDBD'}
          value={value}
        />
        <Text style={styles.valueCurrency}>
          {!usdMode && ' ' + getSymbol(account.currency)}
        </Text>
      </View>

      <TouchableOpacity
        style={{
          width: 30,
          height: 30,
          position: 'absolute',
          right: 15,
          top: 8,
        }}
        onPress={() => {
          setUsdMode(!enterAmountInfo.isUSDMode);
          setValue(String(enterAmountInfo.alternativeValue));
        }}>
        <Image
          source={require('assets/img/change.png')}
          style={{width: 30, height: 30}}
        />
      </TouchableOpacity>

      <TouchableOpacity
        style={{
          width: 30,
          height: 30,
          position: 'absolute',
          left: 15,
          top: 8,
        }}
        onPress={async () => {
          let v = limit;
          setUsdMode(false);
          console.log('v1:' + math.convertFromPlanckToString(v, api.decimals));
          setValue(math.convertFromPlanckToString(v, api.decimals));
          if (!isChatBotLimit) {
            calculateFee(account.address, v).then((fee) => {
              if (fee === undefined) {
                const emptyValue = resetValues(false, 'Invalid connection. Please, try again.'); //TODO go to string
                onChangeValues(emptyValue);
                setEnterAmountInfo(emptyValue);
                setErrorText('Invalid connection. Please, try again.');
                setLoadingEnd(true);
                return;
              }
              v = v.sub(fee);
              setValue(math.convertFromPlanckToString(v, api.decimals));
              console.log('v2:' + math.convertFromPlanckToString(v, api.decimals));
            });
          }
        }}>
        <Image
          source={require('assets/img/max.png')}
          style={{width: 30, height: 30}}
        />
      </TouchableOpacity>
      <View style={[!enterAmountInfo.isValid && !isFeeLoading ? styles.redLine : styles.line, {width: width}]} />
      <View style={{flexDirection: 'row', width: width}}>
        {isFeeLoading ? (
          <ActivityIndicator testID="loader" size={25} color="#2AB2E2" />
        ) : !enterAmountInfo.isValid ? (
          <Text style={styles.errorText}>{errorText}</Text>
        ) : (
          <>
            <View style={{width: '50%', alignItems: 'flex-start'}}>
              {!isChatBotLimit && enterAmountInfo.usdFee !== 0 && (
                <Text style={[styles.subValue]}>
                  {StringUtils.texts.FeeTitle} ${enterAmountInfo.usdFee}
                </Text>
              )}
            </View>
            <View style={{width: '50%', alignItems: 'flex-end'}}>
              {!isFeeLoading && enterAmountInfo.alternativeValue !== 0 && (
                <Text style={styles.subValue}>
                  {usdMode
                    ? `${enterAmountInfo.alternativeValue} ${getSymbol(account.currency)}`
                    : `$${enterAmountInfo.alternativeValue}`}
                </Text>
              )}
            </View>
          </>
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  line: {
    borderBottomWidth: 1,
    borderColor: '#DADADA',
  },
  redLine: {
    borderBottomWidth: 1,
    borderColor: '#EA4335',
  },
  value: {
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
  errorText: {
    marginTop: 5,
    marginLeft: 5,
    fontSize: 14,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: '#EA4335',
  },
  valueCurrency: {
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: '#888888',
  },
  subValue: {
    marginTop: 7,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
});
