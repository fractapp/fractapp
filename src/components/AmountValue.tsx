import React from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import {Currency, getSymbol} from 'types/wallet';

/**
 * Text input for amount
 * @category Components
 */
export const AmountValue = ({
  value,
  alternativeValue,
  fee,
  currency,
  isUSDMode,
  width = '100%',
  onPress,
}: {
  value: number;
  alternativeValue: number;
  fee: number;
  currency: Currency;
  isUSDMode: boolean;
  width?: string;
  onPress?: () => void;
}) => {
  return (
    <TouchableOpacity
      style={{width: width, alignItems: 'center'}}
      onPress={onPress}>
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        <Text style={styles.value}>{isUSDMode && value !== 0 && '$'}</Text>
        <TextInput
          style={[
            styles.value,
            {
              color: value === 0 ? '#BFBDBD' : 'black',
              width: value === 0 ? width : 'auto',
              alignSelf: value === 0 ? 'flex-start' : 'center',
              paddingLeft: 0,
              paddingRight: 0,
              paddingBottom: 0,
              paddingTop: 0,
              marginTop: 8,
              marginBottom: 8,
            },
          ]}
          editable={false}
          value={value === 0 ? 'Enter amount' : String(value)}
          keyboardType={'decimal-pad'}
        />
        <Text style={styles.valueCurrency}>
          {!isUSDMode && value !== 0 && ' ' + getSymbol(currency)}
        </Text>
      </View>

      <View style={[styles.line, {width: width}]} />
      <View style={{flexDirection: 'row', width: width}}>
        <View style={{width: '50%', alignItems: 'flex-start'}}>
          {fee !== 0 && <Text style={[styles.subValue]}>Fee ${fee}</Text>}
        </View>
        <View style={{width: '50%', alignItems: 'flex-end'}}>
          {alternativeValue !== 0 && (
            <Text style={styles.subValue}>
              {isUSDMode
                ? `${alternativeValue} ${getSymbol(currency)}`
                : `$${alternativeValue}`}
            </Text>
          )}
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  line: {
    borderBottomWidth: 1,
    borderColor: '#DADADA',
  },
  value: {
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
  valueCurrency: {
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: '#888888',
  },
  subValue: {
    marginTop: 7,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
});
