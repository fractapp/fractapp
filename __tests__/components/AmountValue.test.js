import React from 'react';
import renderer from 'react-test-renderer';
import {AmountValue} from 'components/AmountValue';
import {Currency} from 'types/wallet';

it('Test one', () => {
  const tree = renderer
    .create(
      <AmountValue
        alternativeValue={555}
        fee={100}
        currency={Currency.Polkadot}
        isUSDMode={true}
        value={100}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it('Test two', () => {
  const tree = renderer
    .create(
      <AmountValue
        alternativeValue={111}
        fee={11}
        currency={Currency.Kusama}
        isUSDMode={false}
        value={100}
        width={'90%'}
        onPress={() => console.log('test')}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it('Test three', () => {
  const tree = renderer
    .create(
      <AmountValue
        alternativeValue={111}
        fee={11}
        currency={Currency.Kusama}
        isUSDMode={false}
        value={0}
        width={'90%'}
        onPress={() => console.log('test')}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
