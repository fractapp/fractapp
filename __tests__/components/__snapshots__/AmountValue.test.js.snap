// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test one 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    >
      $
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={false}
      keyboardType="decimal-pad"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 17,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "alignSelf": "center",
            "color": "black",
            "marginBottom": 8,
            "marginTop": 8,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "auto",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="100"
    />
    <Text
      style={
        Object {
          "color": "#888888",
          "fontFamily": "Roboto-Regular",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#DADADA",
        },
        Object {
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginTop": 7,
            },
          ]
        }
      >
        Fee $
        100
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginTop": 7,
          }
        }
      >
        555 DOT
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test three 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      editable={false}
      keyboardType="decimal-pad"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 17,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "alignSelf": "flex-start",
            "color": "#BFBDBD",
            "marginBottom": 8,
            "marginTop": 8,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "90%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="Enter amount"
    />
    <Text
      style={
        Object {
          "color": "#888888",
          "fontFamily": "Roboto-Regular",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#DADADA",
        },
        Object {
          "width": "90%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginTop": 7,
            },
          ]
        }
      >
        Fee $
        11
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginTop": 7,
          }
        }
      >
        $111
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test two 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      editable={false}
      keyboardType="decimal-pad"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 17,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "alignSelf": "center",
            "color": "black",
            "marginBottom": 8,
            "marginTop": 8,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "auto",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="100"
    />
    <Text
      style={
        Object {
          "color": "#888888",
          "fontFamily": "Roboto-Regular",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    >
       KSM
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#DADADA",
        },
        Object {
          "width": "90%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginTop": 7,
            },
          ]
        }
      >
        Fee $
        11
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginTop": 7,
          }
        }
      >
        $111
      </Text>
    </View>
  </View>
</View>
`;
